pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -128,35 +131,54 @@ func ConvertPolicyToJwtConfig(policy *authn.Policy) *jwtfilter.JwtAuthentication
 		AllowMissingOrFailed: true,
 	}
 	for _, policyJwt := range policyJwts {
-		hostname, port, _, err := model.ParseJwksURI(policyJwt.JwksUri)
-		if err != nil {
-			log.Errorf("Cannot parse jwks_uri %q: %v", policyJwt.JwksUri, err)
-			continue
-		}
-
 		jwt := &jwtfilter.JwtRule{
-			Issuer:    policyJwt.Issuer,
-			Audiences: policyJwt.Audiences,
-			JwksSourceSpecifier: &jwtfilter.JwtRule_RemoteJwks{
-				RemoteJwks: &jwtfilter.RemoteJwks{
-					HttpUri: &core.HttpUri{
-						Uri: policyJwt.JwksUri,
-						HttpUpstreamType: &core.HttpUri_Cluster{
-							Cluster: JwksURIClusterName(hostname, port),
-						},
-					},
-					CacheDuration: &types.Duration{Seconds: jwtPublicKeyCacheSeconds},
-				},
-			},
+			Issuer:               policyJwt.Issuer,
+			Audiences:            policyJwt.Audiences,
 			ForwardPayloadHeader: OutputLocationForJwtIssuer(policyJwt.Issuer),
 			Forward:              true,
 		}
+
 		for _, location := range policyJwt.JwtHeaders {
 			jwt.FromHeaders = append(jwt.FromHeaders, &jwtfilter.JwtHeader{
 				Name: location,
 			})
 		}
 		jwt.FromParams = policyJwt.JwtParams
+
+		if enableLocalJwks {
+			jwtPubKey, err := model.ResolveJwtPubKey(policyJwt.Issuer)
```

Comment:
Not all JWT using openID. In other words, if policyJwt defined jwksURI, you should use that to fetch public key instead.

Reply:
done